crates/aptos-genesis/src/config.rs

```diff
@@ -96,13 +96,17 @@ impl Default for Layout {
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct ValidatorConfiguration {
     /// Account address
-    pub account_address: AccountAddress,
+    pub owner_account_address: AccountAddress,
+    /// Key used for signing transactions with the account
+    pub owner_account_public_key: Ed25519PublicKey,
+    pub operator_account_address: AccountAddress,
+    pub operator_account_public_key: Ed25519PublicKey,
+    pub voter_account_address: AccountAddress,
+    pub voter_account_public_key: Ed25519PublicKey,
```

Comment:
It seems very odd to me to have voter in the ValidatorConfiguration.

Maybe we should create this in a nested way:
```

pub struct StakePoolConfiguration {
    /// Account address
    pub owner_account_address: AccountAddress,
    /// Key used for signing transactions with the account
    pub owner_account_public_key: Ed25519PublicKey,
    pub operator_account_address: AccountAddress,
    pub operator_account_public_key: Ed25519PublicKey,
    pub voter_account_address: AccountAddress,
    pub voter_account_public_key: Ed25519PublicKey,
    /// Stake amount for consensus
    pub stake_amount: u64,
    pub validator_node_configuration: ValidatorNodeConfiguration
}

pub struct ValidatorNodeConfiguration {
    /// maybe have this field, or maybe skip it as redundant:
    pub pool_address: AccountAddress,
    /// Key used for signing in consensus
    pub consensus_public_key: bls12381::PublicKey,
    /// Corresponding proof of possession of consensus public key
    pub proof_of_possession: bls12381::ProofOfPossession,
    /// Key used for signing transactions with the account
    pub account_public_key: Ed25519PublicKey,
    /// Public key used for validator network identity (same as account address)
    pub validator_network_public_key: x25519::PublicKey,
    /// Host for validator which can be an IP or a DNS name
    pub validator_host: HostAndPort,
    /// Public key used for full node network identity (same as account address)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub full_node_network_public_key: Option<x25519::PublicKey>,
    /// Host for full node which can be an IP or a DNS name and is optional
    #[serde(skip_serializing_if = "Option::is_none")]
    pub full_node_host: Option<HostAndPort>,
}
```


I assume these are just CLI/test configurations, the real transaction in genesis is in aptos-move/vm-genesis/src/lib.rs, and that also has Validator instead of a StakePool?

Reply:
This is matching the `Validator` object from onchain.  So, we could rename it, but I'm leaving it like that for now.